test_runner/regress/test_walredo_not_left_behind_on_detach.py

```diff
@@ -0,0 +1,106 @@
+import time
+from pathlib import Path
+
+import psutil
+import pytest
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import NeonEnvBuilder, PageserverApiException
+from fixtures.types import TenantId
+
+
+def assert_child_processes(pageserver_pid, wal_redo_present=False, defunct_present=False):
+    # XXX --ppid is probably GNU-only
+    children = list(psutil.Process(pageserver_pid).children())
+    for child in children:
+        if not wal_redo_present:
+            assert "--wal-redo" not in list(child.cmdline())
+        if not defunct_present:
+            assert child.status() != psutil.STATUS_ZOMBIE
+
+
+# Check that the pageserver doesn't leave behind WAL redo processes
+# when a tenant is detached. We had an issue previously where we failed
+# to wait and consume the exit code of the WAL redo process, leaving it behind
+# as a zombie process.
+def test_walredo_not_left_behind_on_detach(neon_env_builder: NeonEnvBuilder):
+    env = neon_env_builder.init_start()
+    pageserver_http = env.pageserver.http_client()
+
+    pagserver_pid = int((Path(env.repo_dir) / "pageserver.pid").read_text())
+
+    assert_child_processes(pagserver_pid, wal_redo_present=False, defunct_present=False)
+
+    # first check for non existing tenant
+    tenant_id = TenantId.generate()
+    with pytest.raises(
+        expected_exception=PageserverApiException,
+        match=f"Tenant not found for id {tenant_id}",
+    ):
+        pageserver_http.tenant_detach(tenant_id)
+
+    # create new nenant
+    tenant_id, timeline_id = env.neon_cli.create_tenant()
```

Comment:
We don't use this timeline_id right?

```suggestion
    tenant_id, _timeline_id = env.neon_cli.create_tenant()
```